remoulade/state/middleware.py

```diff
@@ -20,33 +21,41 @@ def __init__(self, backend, state_ttl=3600, max_size=2e6):
         self.state_ttl = state_ttl
         self.max_size = max_size
 
-    def save(self, message, state):
-        args = message.args
-        kwargs = message.kwargs
+    def save(self, message, nameState, **times):
+        args = str(message.args)
+        kwargs = str(message.kwargs)
         message_id = message.message_id
         if sys.getsizeof(args) > self.max_size:
             # Arguments exceed maximum size to display
             #  do not save them.
-            args = []
+            args = "[]"
         if sys.getsizeof(kwargs) > self.max_size:
             # Keyword arguments exceed maximum size to
             #  display do not save them
-            kwargs = {}
-        self.backend.set_state(State(message_id, state, args, kwargs), self.state_ttl)
+            kwargs = "{}"
+        objState = State(message_id, nameState, args, kwargs, **times)
+        self.backend.set_state(objState, self.state_ttl)
+
+    def _get_current_time(self):
+        return datetime.now().strftime("%Y/%m/%d-%H:%M:%S")
```

Comment:
Always better to set a timezone when using dates: `datetime.now(timezone.utc)`. Also the conversion to string could be done is the asdict method of state (we could want to use the date somewhere). Also use .isoformat() function to format it as it's a norm (https://docs.python.org/fr/3/library/datetime.html#datetime.datetime.isoformat).